app/lib/katello/resources/discovery/file_discovery.rb

```diff
@@ -0,0 +1,34 @@
+module Katello
+  class FileDiscovery < RepoDiscovery
+    attr_reader :found, :crawled, :to_follow
+    def initialize(url, upstream_username = nil,
+                   upstream_password = nil, search = '*', crawled = [],
+                   found = [], to_follow = [])
+      @uri = uri(url)
+      @upstream_username = upstream_username.empty? ? nil : upstream_username
+      @upstream_password = upstream_password.empty? ? nil : upstream_password
+      @search = search
+      @found = found
+      @crawled = crawled
+      @to_follow = to_follow
+    end
+
+    def run(resume_point)
+      file_crawl(resume_point)
+    end
+
+    private
+
+    def file_crawl(resume_point)
```

Comment:
Since there is no recursion in the method, perhaps just make this `run(resume_point)`?

Reply:
Updated to remove this extra method.